api/library/python/iterm2/iterm2/alert.py

```diff
@@ -146,3 +147,165 @@ async def async_run(
              f'placeholder: {placeholder}, ' +
              f'defaultValue: {default_value}, ' +
              f'window_id: {json.dumps(self.window_id)})'))
+
+
+class PolyModalAlert:
+    """A modal alert with various ui options
+
+    :param title: The title, shown in bold at the top.
+    :param subtitle: The informative text, which may be more than one line
+        long.
+    :param window_id: The window to attach the alert to. If None, it will be
+        application modal."
+    """
+
+    # pylint: disable=too-many-instance-attributes
+
+    def __init__(
+            self,
+            title: str,
+            subtitle: str,
+            window_id: typing.Optional[str] = None):
+        self.__title: str = title
+        self.__subtitle: str = subtitle
+        self.__buttons: typing.List[str] = []
+        self.__checkboxes: typing.List[str] = []
+        self.__checkbox_defaults: typing.List[int] = []
+        self.__combobox_items: typing.List[str] = []
+        self.__combobox_default: str = ""
+        self.__text_field: typing.List[str] = []
+        self.__width: int = 300
+        self.__window_id = window_id
+
+    @property
+    def title(self) -> str:
+        """Returns the title string"""
+        return self.__title
+
+    @property
+    def subtitle(self) -> str:
+        """Returns the subtitle string"""
+        return self.__subtitle
+
+    @property
+    def text_field(self) -> typing.List[str]:
+        """Returns the text field list [placeholder, default] """
+        return self.__text_field
+
+    @property
+    def checkboxes(self) -> typing.List[str]:
+        """Returns a list of checkbox strings """
+        return self.__checkboxes
+
+    @property
+    def checkbox_defaults(self) -> typing.List[int]:
+        """Returns a list of whether checkboxes are checked by default """
+        return self.__checkbox_defaults
+
+    @property
+    def combobox_items(self) -> typing.List[str]:
+        """Returns the list of strings that will populate the combobox"""
+        return self.__combobox_items
+
+    @property
+    def combobox_default(self) -> str:
+        """Returns the combox item that is initially chosen"""
+        return self.__combobox_default
+
+    @property
+    def window_id(self) -> typing.Optional[str]:
+        """Returns the window ID"""
+        return self.__window_id
+
+    def add_button(self, title: str):
+        """Adds a button to the end of the list of buttons."""
+        self.__buttons.append(title)
+
+    def add_checkboxes(self, items: typing.List[str],
+                       defaults: typing.List[int]):
+        """Adds an array of checkbox items to the end of the list of
```

Comment:
Update docstring to describe arguments and return value. That way the generated documentation will be more useful. Same for other methods of this class that take parameters.